packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts

```diff
@@ -0,0 +1,252 @@
+import { Web3Wrapper } from '@0x/dev-utils';
+import { TakerRequestQueryParams, V4RFQFirmQuote, V4RFQIndicativeQuote } from '@0x/quote-server';
+import { BigNumber, logUtils } from '@0x/utils';
+import { AxiosInstance } from 'axios';
+
+import {
+    AltFirmQuoteResponse,
+    AltIndicativeQuoteReponse,
+    AltOffering,
+    AltQuoteModel,
+    AltQuoteRequestData,
+    AltQuoteSide,
+    AltRfqtMakerAssetOfferings,
+} from '../types';
+
+const IMPUTED_EXPIRY_SECONDS = 120;
+
+function getAltMarketInfo(
+    offerings: AltOffering[],
+    buyTokenAddress: string,
+    sellTokenAddress: string,
+): AltOffering | undefined {
+    for (const offering of offerings) {
+        if (
+            (buyTokenAddress === offering.baseAsset && sellTokenAddress === offering.quoteAsset) ||
+            (sellTokenAddress === offering.baseAsset && buyTokenAddress === offering.quoteAsset)
+        ) {
+            return offering;
+        }
+    }
+    return undefined;
+}
+
+function parseFirmQuoteResponseFromAltMM(altFirmQuoteReponse: AltFirmQuoteResponse): V4RFQFirmQuote {
+    return {
+        signedOrder: altFirmQuoteReponse.data['0xv4order'],
+    };
+}
+
+function parseIndicativeQuoteResponseFromAltMM(
+    altIndicativeQuoteReponse: AltIndicativeQuoteReponse,
+    altPair: AltOffering,
+    makerToken: string,
+    takerToken: string,
+): V4RFQIndicativeQuote {
+    let makerAmount: BigNumber;
+    let takerAmount: BigNumber;
+
+    if (!altIndicativeQuoteReponse.price) {
+        throw new Error('Price not returned by alt MM');
+    } else {
+        if (makerToken === altPair.baseAsset) {
+            if (altIndicativeQuoteReponse.amount) {
+                // base asset = maker asset, quote asset = taker asset, amount specified
+                makerAmount = Web3Wrapper.toBaseUnitAmount(
+                    new BigNumber(altIndicativeQuoteReponse.amount),
```

Comment:
There is quite a bit of logic here.. can you add some comments explaining the difference between `amount` and `value`?

Reply:
added some comments 👍